core/src/main/java/io/confluent/rest/metrics/MetricsResourceMethodApplicationListener.java

```diff
@@ -256,7 +319,24 @@ public void finished(long requestSize, long responseSize, long latencyMs) {
     /**
      * Indicate that a request has failed with an exception.
      */
-    public void exception() {
+    public void exception(final RequestEvent event) {
+      int errorCode = -1;
+      if (event.getException() instanceof RestException) {
```

Comment:
why not read the status of the response in the event? In which case, we would never have an unknown scenario.